### PR TITLE
Feature: Catchup Eval Stake Exception Round Handling

### PR DESCRIPTION
**Pull request by gmalouf**
## Summary

We had a brief period where a handful of rounds had stake miniscully mismatching in some cases based on delta corruption in some nodes. This pull request ensures catchup cannot stall while removing the stake dilution factor (underlying cause/corrections were addressed in the last consensus upgrade).

## Test Plan

Added a new `TestEval_EndOfBlockStake` (thanks @algorandskiy ) to exercise various cases being handled as expected.

### CODE REVIEW

File: ledger/eval/eval.go [hunk 1]
```diff
@@ -1715,11 +1708,11 @@ transactionGroupLoop:
...
 		case <-ctx.Done():
 			return ledgercore.StateDelta{}, ctx.Err()
-		case err, open := <-txvalidator.done:
+		case err1, open := <-txvalidator.done:
 			if !open {
 				break
 			}
-			if err != nil {
+			if err1 != nil {
 				return ledgercore.StateDelta{}, err
```
Comment (algorandskiy) [inline ledger/eval/eval.go:1716]:
```suggestion
				return ledgercore.StateDelta{}, err1
```

File: ledger/eval/eval_test.go [hunk 2]
```diff
@@ -1445,3 +1401,179 @@ func TestExpiredAccountGeneration(t *testing.T) {
...
+	require.NoError(t, digestError)
+
+	var tests = []struct {
+		actual      basics.MicroAlgos
+		expected    basics.MicroAlgos
+		round       basics.Round
+		genesisHash crypto.Digest
+		err         string
+	}{
+		// Normal round mainnet, matched stake => no error
```
Comment (algorandskiy) [inline ledger/eval/eval_test.go:1419]: why mainnet?

File: ledger/eval/eval.go [hunk 3]
```diff
@@ -1318,29 +1318,22 @@ func (eval *BlockEvaluator) endOfBlock() error {
...
+
+		if eval.block.StateProofTracking[protocol.StateProofBasic].StateProofOnlineTotalWeight != expectedVotersWeight {
+			mainnetGenesisHash, _ := crypto.DigestFromString("YBQ4JWH4DW655UWXMBF6IVUOH5WQIGMHVQ333ZFWEC22WOJERLPQ")
+			if eval.genesisHash == mainnetGenesisHash {
+				// Handful of historic rounds where the state proof online total weight is known to be slightly different
+				// than expected voters weight. A consensus release (V38) addressed this/prevented the corner case from
+				// occurring going forward.
+				switch eval.block.Round() {
+				case 24018688, 26982912, 27009024, 27713280, 27822080, 27822848, 27929344, 28032768,
+					28977920, 29822208, 30005248, 30033920:
```
Comment (algorandskiy) [inline ledger/eval/eval.go:1330]: rounds 30M were like 3-4 months ago but the fix was like I think since April?

Comment (algorandskiy) [reply ledger/eval/eval.go:1330]: talked offline, this was gated by consensus param

File: ledger/eval/eval_test.go (hunk 2 above)
Comment (algorandskiy) [reply ledger/eval/eval_test.go:1419]: talked offline, this  has explicit round numbers to actually hit newly added switch-case for mainnet.

File: ledger/eval/eval_test.go [hunk 4]
```diff
@@ -1445,3 +1401,179 @@ func TestExpiredAccountGeneration(t *testing.T) {
...
+				block:       newBlock,
+				l:           l,
+			}
+
+			err = eval.endOfBlock()
+			if len(test.err) == 0 {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), test.err)
```
Comment (jasonpaulos) [inline ledger/eval/eval_test.go:1574]: nit/fyi: `require.ErrorContains` could be used to consoldate these two checks

File: ledger/eval/eval.go (hunk 1 above)
Comment (jasonpaulos) [reply ledger/eval/eval.go:1716]: +1

Comment (gmalouf) [reply ledger/eval/eval.go]: Addresses in latest, this was what caused test failure too (so good that was picked up automatically!).

File: ledger/eval/eval_test.go (hunk 4 above)
Comment (gmalouf) [reply ledger/eval/eval_test.go]: Agreed!

**Review by algorandskiy — APPROVED** [review]

**Review by jasonpaulos — APPROVED** [review]

**Pull request merged**